### PR TITLE
Added OpenBSD support

### PR DESCRIPTION
**Pull request by rubenllorente**
Added OpenBSD support. Includes a fix for ensuring
lrzip won't try to use more memory than the OS is
willing to grant access to by default

### CODE REVIEW

File: lrzip_private.h [hunk 1]
```diff
@@ -46,8 +46,6 @@
 
 #ifdef HAVE_ALLOCA_H
 # include <alloca.h>
-#elif defined __GNUC__
```
Comment (ckolivas) [inline lrzip_private.h:49]: Why was this removed?

Comment (rubenllorente) [reply lrzip_private.h:49]: My bad. That's a leftover from local testing. Since OpenBSD does use Clang as a default compiler in amd64 I didn't detect it during testing.

I just built and tested this set of patches without lines 49-50 of this PR and it worked fine. Sorry for the noise :)

Comment (ckolivas) [reply lrzip_private.h:49]: Thanks. Do you want to respin the patch or shall I manually reinsert it?

**Comment by rubenllorente** [comment]
I am closing this PR and attempting to generate one that will be valid directly.
